drools-ansible-rulebook-integration-protoextractor/src/main/java/org/drools/ansible/rulebook/integration/protoextractor/prototype/ValueCollectVisitor.java

```diff
@@ -0,0 +1,133 @@
+package org.drools.ansible.rulebook.integration.protoextractor.prototype;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.drools.ansible.rulebook.integration.protoextractor.ast.ASTNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.DefaultedVisitor;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.ExtractorNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.IdentifierNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.IndexAccessorNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.SquaredAccessorNode;
+import org.kie.api.prototype.Prototype;
+
+public class ValueCollectVisitor extends DefaultedVisitor<Object> {
+    private final Object original;
+    private Object cur; // cur doesn't have to one Node in the event, it can be List of Nodes matching the path
+
+    public ValueCollectVisitor(Object original) {
+        this.original = original;
+        this.cur = this.original;
+    }
+
+    @Override
+    public Object defaultVisit(ASTNode n) {
+        throw new UnsupportedOperationException("this visitor implemented all visit methods");
+    }
+
+    @Override
+    public Object visit(IdentifierNode n) {
+        if (cur instanceof List) {
+            cur = collectPathsForAllElementsInArray(n);
+        } else {
+            cur = fromMap(cur, n.getValue());
+        }
+        return cur;
+    }
+
+    private Object collectPathsForAllElementsInArray(IdentifierNode n) {
+        // if accessing a list without an index, evaluate all elements and collect matching children
+        List<?> theList = (List<?>) cur;
+        List<Object> nextNodeList = new PathWrapperList();
```

Comment:
When a visitor faces array access without brackets (e.g. `alerts`), I introduce `PathWrapperList` to broaden the search paths. I haven't yet tested a case where `PathWrapperList` is nested (probably I need to strip it).